internal/api/provider/azure.go

```diff
@@ -39,11 +39,16 @@ type azureProvider struct {
 }
 
 var azureIssuerRegexp = regexp.MustCompile("^https://login[.]microsoftonline[.]com/([^/]+)/v2[.]0/?$")
+var azureCIAMIssuerRegexp = regexp.MustCompile("^https://[a-z0-9-]+[.]ciamlogin[.]com/([^/]+)/v2[.]0/?$")
```

Comment:
/nit 

It's an edge case but is there any restriction around tenant/subdomain naming?

It's rare but if the subdomain includes an underscore (e.g. `https://supabase_auth.ciamlogin.com/` it won't match

Reply:
Seems like [letters and numbers only](https://techcommunity.microsoft.com/t5/microsoft-365/tenant-name-restrictions-in-o365/m-p/638184)